src/ECS/Systems/PathfindingSystem.cpp

```diff
@@ -28,43 +39,557 @@ PathfindingSystem& PathfindingSystem::instance()
+	    {cellIndex.x - 1, cellIndex.y + 1}, // Under and left
+	    {cellIndex.x + 1, cellIndex.y - 1}, // Over and right
+	    {cellIndex.x + 1, cellIndex.y + 1}, // Under and right
+	};
+}
+
+/// Iterate between all adjacent grids and find closest object that the ray (step) intersects with (circle)
+/// If that object is in front (and we are not in it) and less than 256 steps away, set as target and store steps
+bool LinearScanForObstacle(entt::entity entity, const glm::vec2& pos, const glm::vec2& step)
+{
+	const auto& map = Game::instance()->GetEntityMap();
+	auto& registry = Game::instance()->GetEntityRegistry();
+
+	// Reference will be updated or removed
+	registry.Remove<WallHugObjectReference>(entity);
+
+	// FIXME(bwrsandman): This gets first, not closest
+	std::optional<entt::entity> fixedEntity = std::nullopt;
+	for (const auto& c : GetNeighboringCells(pos + step))
+	{
+		// TODO(bwrsandman): Skip if out of bounds or in water
+		const auto& fixed = map.GetFixedInGridCell(c);
+		if (!fixed.empty())
+		{
+			auto iter = std::find_if(fixed.cbegin(), fixed.cend(), [&registry](const auto& f) {
+				return !registry.AnyOf<Field>(f); // TODO && registry.AllOf<CollideData>();
+			});
+			if (iter != fixed.cend())
+			{
+				fixedEntity = std::make_optional(*iter);
+				break;
+			}
+		}
+	}
+	if (!fixedEntity.has_value())
+	{
+		return false;
+	}
+
+	// Do ray-circle intersection with all objects found
+	// TODO: list of neighbouring cells?
+	const auto& fixed = registry.Get<Fixed>(*fixedEntity);
+	const auto stepSize = glm::length(step);
+	const auto direction = step / stepSize;
+	// Do a ray-circle intersection in 2d with ray = {pos, normal}, circle = {fixed.c, fixed.r} (same as ray-sphere)
+	const auto oc = pos - fixed.boundingCenter;
+	const auto halfB = glm::dot(oc, direction);
+	const auto c = glm::length2(oc) - fixed.boundingRadius * fixed.boundingRadius;
+	const float discriminant = halfB * halfB - c;
+	const bool hit = discriminant > 0;
+
+	// TODO: if no hit, go through neighbouring cells, if still none, remove WallHugReference and return 1
+	if (!hit)
+	{
+		return false;
+	}
+
+	const float t = hit ? -halfB - glm::sqrt(discriminant) : -1.0f;
+	const bool inFront = t > 0.0f;
+
+	if (!inFront)
+	{
+		return false;
+	}
+
+	const auto numSteps = t / stepSize;
+	assert(numSteps >= 0); // t wouldn't be positive here and size should always be positive
+
+	// Too far
+	if (numSteps >= std::numeric_limits<decltype(WallHugObjectReference::stepsAway)>::max())
+	{
+		return false;
+	}
+
+	// Store object and number of steps away
+	registry.Assign<WallHugObjectReference>(entity, static_cast<decltype(WallHugObjectReference::stepsAway)>(numSteps),
+	                                        *fixedEntity);
+	return true;
+}
+
+/// If we have a number of turns to an obstacle but no obstacle saved: search in an arc to find one
+bool OrbitScanForObstacle(entt::entity entity, bool clockwise, Transform& transform, WallHug& wallHug)
```

Comment:
Clang-Tidy: Function 'OrbitScanForObstacle' is within a recursive call chain
[misc-no-recursion]

Reply:
Should we avoid recursive calls or disable the check?
This would be the only place there is one.
I could rework it as a limited loop.